### PR TITLE
feature(client): add google maps and places api and location autocomplete

### PR DESCRIPTION
**Pull request by thnkloud9**
Still need to

- [x] resolve issue importing angular config (for google apikey)
- [x] add action for saveLocation before saveEvent
- [x] add form validation for location
- [x] add unit tests

### CODE REVIEW

**Comment by ioanpocol** [comment]
Superdesk already has a controlled vocabulary with locations/places. I think you have to use that list here.

**Comment by thnkloud9** [comment]
@ioanpocol , this will be for planning component where we do not want to limit location to controlled vocabulary.

**Comment by ioanpocol** [comment]
Please ask Karel about this, just to be sure about your asumptions.

File: client/styles/index.scss [hunk 1]
```diff
@@ -3,6 +3,7 @@
 @import './events-list.scss';
 @import './add-event-container.scss';
 @import './add-event-form.scss';
+@import './react-geosuggest.scss';
```
Comment (vied12) [inline client/styles/index.scss]: you should import this where you are importing the geosuggest module, in `AddGeoSuggestInput.jsx` (as I did with the timepicker)

Comment (thnkloud9) [inline client/styles/index.scss]: this is overrides for our implementation (this css in react-geosuggest/module is only for the example for that component).

**Pull request merged**